tests/testutils.hpp

```diff
@@ -115,7 +115,7 @@ struct QmlBot {
   QmlBot() {
     qquick_view = new QQuickView();
     auto qmltester = fs::path(__FILE__).parent_path() / "qmltester.qml";
-    qquick_view->setSource(QUrl{qmltester.c_str()});
```

Comment:
I assume that this cause
> [also there were two minor issues with testutils.cpp not critical intiailizers that I've commented, we can fix it whenever later](https://discord.com/channels/@me/1116356243200671745/1143305731660062741)
Can you explain what is "critical intiailizers"?

Reply:
It was throwing compile time error at my end, I'm using Qt 6.5.0 installed with aqt installer but it seems initializing with enclosed braces not supported in this version or something depended on my platform machine.
the error was:
```
qtgql\tests\testutils.hpp:118:50: error: no matching function for call to 'QUrl::QUrl(<brace-enclosed initializer list>)'
  118 |     qquick_view->setSource(QUrl{qmltester.c_str()});
  
  C:/Users/Mortie/MyConnandeps/Qt/6.5.0/mingw_64/include/QtCore/qurl.h:151:25: note:   no known conversion for argument 1 from 'const std::filesystem::__cxx11::path::value_type*' {aka 'const wchar_t*'} to 'const QString&'
  151 |     QUrl(const QString &url, ParsingMode mode = TolerantMode);
```

Reply:
gotcha